### PR TITLE
Enable nullability in WebBrowserBase

### PR DESCRIPTION
**Pull request by gpetrou**
## Proposed changes

- Enable nullability in WebBrowserBase.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/9350)

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/WebBrowserContainer.cs [hunk 1]
```diff
@@ -345,7 +345,7 @@ internal void RemoveControl(Control ctl)
                 return ctl.container;
             }
 
-            ScrollableControl f = ctl.ContainingControl;
+            ScrollableControl? f = ctl.ContainingControl;
```
Comment (elachlan) [inline src/System.Windows.Forms/src/System/Windows/Forms/WebBrowserContainer.cs:348]: Need a better variable name.

File: src/System.Windows.Forms/src/System/Windows/Forms/WebBrowserContainer.cs [hunk 2]
```diff
@@ -376,13 +376,13 @@ internal void OnUIActivate(WebBrowserBase site)
...
             WebBrowserBase tempSite = siteUIActive;
-            tempSite.AXInPlaceObject.UIDeactivate();
+            tempSite.AXInPlaceObject?.UIDeactivate();
         }
 
         site.AddSelectionHandler();
         Debug.Assert(siteUIActive is null, "Object did not call OnUIDeactivate");
         siteUIActive = site;
-        ContainerControl f = site.ContainingControl;
+        ContainerControl? f = site.ContainingControl;
```
Comment (elachlan) [inline src/System.Windows.Forms/src/System/Windows/Forms/WebBrowserContainer.cs:385]: Same here.

File: src/System.Windows.Forms/src/System/Windows/Forms/WebBrowserBase.cs [hunk 3]
```diff
@@ -39,23 +37,23 @@ public unsafe partial class WebBrowserBase : Control
...
-    private WebBrowserSiteBase axSite;
-    private ContainerControl containingControl;
+    private WebBrowserSiteBase? axSite;
+    private ContainerControl? containingControl;
     private HWND hwndFocus;
-    private EventHandler selectionChangeHandler;
+    private EventHandler? selectionChangeHandler;
     private readonly Guid clsid;
     // Pointers to the ActiveX object: Interface pointers are cached for perf.
-    private IOleObject.Interface axOleObject;
```
Comment (dreddy-work) [inline src/System.Windows.Forms/src/System/Windows/Forms/WebBrowserBase.cs:48]: Can you fix naming violations in this file?

File: src/System.Windows.Forms/src/System/Windows/Forms/WebBrowserBase.cs [hunk 4]
```diff
@@ -1174,7 +1166,7 @@ private unsafe Size SetExtent(int width, int height)
     private unsafe Size GetExtent()
     {
         Size size = default;
-        axOleObject.GetExtent(DVASPECT.DVASPECT_CONTENT, (SIZE*)&size);
+        axOleObject?.GetExtent(DVASPECT.DVASPECT_CONTENT, (SIZE*)&size);
```
Comment (dreddy-work) [inline src/System.Windows.Forms/src/System/Windows/Forms/WebBrowserBase.cs:1169]: why not `!`  ?

File: src/System.Windows.Forms/src/System/Windows/Forms/WebBrowserBase.cs (hunk 3 above)
Comment (dreddy-work) [reply src/System.Windows.Forms/src/System/Windows/Forms/WebBrowserBase.cs:48]: follow-up PR should fine.

File: src/System.Windows.Forms/src/System/Windows/Forms/WebBrowserBase.cs [hunk 5]
```diff
@@ -1066,19 +1058,19 @@ internal WebBrowserSiteBase ActiveXSite
     {
         get
         {
-            axSite ??= CreateWebBrowserSiteBase();
+            _axSite ??= CreateWebBrowserSiteBase();
```
Comment (dreddy-work) [inline src/System.Windows.Forms/src/System/Windows/Forms/WebBrowserBase.cs:1061]: nit: convert to an expressional body.

File: src/System.Windows.Forms/src/System/Windows/Forms/WebBrowserBase.cs [hunk 6]
```diff
@@ -1103,20 +1095,20 @@ private EventHandler SelectionChangeHandler
     {
         get
         {
-            selectionChangeHandler ??= new EventHandler(OnNewSelection);
+            _selectionChangeHandler ??= new EventHandler(OnNewSelection);
```
Comment (dreddy-work) [inline src/System.Windows.Forms/src/System/Windows/Forms/WebBrowserBase.cs:1098]: same here.

File: src/System.Windows.Forms/src/System/Windows/Forms/WebBrowserBase.cs [hunk 7]
```diff
@@ -1207,16 +1199,16 @@ private bool EditMode
     {
         get
         {
-            return axEditMode != WebBrowserHelper.AXEditMode.None;
+            return _axEditMode != WebBrowserHelper.AXEditMode.None;
```
Comment (dreddy-work) [inline src/System.Windows.Forms/src/System/Windows/Forms/WebBrowserBase.cs:1202]: same here and also check other places in this file.

File: src/System.Windows.Forms/src/System/Windows/Forms/WebBrowserBase.cs [hunk 8]
```diff
@@ -1227,8 +1219,8 @@ internal ContainerControl FindContainerControlInternal()
             }
         }
 
-        ContainerControl cc = null;
-        for (Control control = this; control is not null; control = control.ParentInternal)
+        ContainerControl? cc = null;
```
Comment (dreddy-work) [inline src/System.Windows.Forms/src/System/Windows/Forms/WebBrowserBase.cs:1222]: `cc ->containerControl`

File: src/System.Windows.Forms/src/System/Windows/Forms/WebBrowserContainer.cs [hunk 9]
```diff
@@ -173,7 +173,7 @@ private void ListAXControls(List<object> list, bool fuseOcx)
             {
                 if (fuseOcx)
                 {
-                    object ax = webBrowserBase._activeXInstance;
+                    object? ax = webBrowserBase._activeXInstance;
```
Comment (dreddy-work) [inline src/System.Windows.Forms/src/System/Windows/Forms/WebBrowserContainer.cs:176]: `ax -> activeX`

**Review by dreddy-work — CHANGES_REQUESTED** [review]

**Comment by gpetrou** [comment]
@dreddy-work any more feedback on this one?

File: src/System.Windows.Forms/src/System/Windows/Forms/WebBrowserBase.cs [hunk 10]
```diff
@@ -223,7 +215,7 @@ internal override unsafe void OnBoundsUpdate(int x, int y, int width, int height
 
     protected override bool ProcessDialogKey(Keys keyData)
     {
-        return ignoreDialogKeys ? false : base.ProcessDialogKey(keyData);
+        return _ignoreDialogKeys ? false : base.ProcessDialogKey(keyData);
```
Comment (dreddy-work) [inline src/System.Windows.Forms/src/System/Windows/Forms/WebBrowserBase.cs:218]: nit: move to expression body.

File: src/System.Windows.Forms/src/System/Windows/Forms/WebBrowserBase.cs [hunk 11]
```diff
@@ -599,24 +591,18 @@ protected override void Dispose(bool disposing)
...
-            axState = value;
-        }
+        get => _axState;
+        set => _axState = value;
     }
 
     internal bool GetAXHostState(int mask)
     {
-        return axHostState[mask];
+        return _axHostState[mask];
```
Comment (dreddy-work) [inline src/System.Windows.Forms/src/System/Windows/Forms/WebBrowserBase.cs:600]: same here.

File: src/System.Windows.Forms/src/System/Windows/Forms/WebBrowserBase.cs [hunk 12]
```diff
@@ -731,25 +717,25 @@ internal unsafe bool DoVerb(OLEIVERB verb)
...
 
-            return containingControl;
+            return _containingControl;
         }
     }
 
     internal WebBrowserContainer CreateWebBrowserContainer()
     {
-        wbContainer ??= new WebBrowserContainer(this);
+        _wbContainer ??= new WebBrowserContainer(this);
```
Comment (dreddy-work) [inline src/System.Windows.Forms/src/System/Windows/Forms/WebBrowserBase.cs:736]: move to expression body.

File: src/System.Windows.Forms/src/System/Windows/Forms/WebBrowserBase.cs [hunk 13]
```diff
@@ -1054,31 +1034,24 @@ private void TransitionFromUIActiveToInPlaceActive()
...
-    {
-        get
-        {
-            axSite ??= CreateWebBrowserSiteBase();
-
-            return axSite;
-        }
-    }
+    internal WebBrowserSiteBase ActiveXSite =>
+        _axSite ??= CreateWebBrowserSiteBase();
```
Comment (dreddy-work) [inline src/System.Windows.Forms/src/System/Windows/Forms/WebBrowserBase.cs:1046]: nice!.

**Review by dreddy-work — CHANGES_REQUESTED** [review]
RLGTM.

**Review by dreddy-work — APPROVED** [review]

**Pull request merged**